akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala

```diff
@@ -257,7 +257,8 @@ object RejectionHandler {
       }
       .handleAll[UnsupportedRequestContentTypeRejection] { rejections ⇒
         val supported = rejections.flatMap(_.supported).mkString(" or ")
-        rejectRequestEntityAndComplete((UnsupportedMediaType, "The request's Content-Type is not supported. Expected:\n" + supported))
+        val unsupportedContentType = rejections.find(_.contentType.isDefined).map(_.contentType)
```

Comment:
This will be an `Option[Option[ContentType]]` now. I think you can just do a `rejections.flatMap(_.contentType)` instead.